server/src/main/java/org/elasticsearch/http/HttpRouteStats.java

```diff
@@ -0,0 +1,186 @@
+            }
+            builder.field("count", histogram[i]);
+            builder.endObject();
+            remainingCount -= histogram[i];
+        }
+        builder.endArray();
+    }
+
+    public static HttpRouteStats merge(HttpRouteStats first, HttpRouteStats second) {
+        assert first.requestSizeHistogram.length == second.requestSizeHistogram.length
+            && first.responseSizeHistogram.length == second.responseSizeHistogram.length
+            && first.responseTimeHistogram.length == second.responseTimeHistogram.length;
+
+        return new HttpRouteStats(
+            first.requestCount + second.requestCount,
+            first.totalRequestSize + second.totalRequestSize,
+            IntStream.range(0, first.requestSizeHistogram.length)
+                .mapToLong(i -> first.requestSizeHistogram[i] + second.requestSizeHistogram[i])
+                .toArray(),
+            first.responseCount + second.responseCount,
+            first.totalResponseSize + second.totalResponseSize,
+            IntStream.range(0, first.responseSizeHistogram.length)
+                .mapToLong(i -> first.responseSizeHistogram[i] + second.responseSizeHistogram[i])
+                .toArray(),
+            IntStream.range(0, first.responseTimeHistogram.length)
+                .mapToLong(i -> first.responseTimeHistogram[i] + second.responseTimeHistogram[i])
+                .toArray()
+        );
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeVLong(requestCount);
+        out.writeVLong(totalRequestSize);
+        out.writeVLongArray(requestSizeHistogram);
+        out.writeVLong(responseCount);
+        out.writeVLong(totalResponseSize);
+        out.writeVLongArray(responseSizeHistogram);
+        out.writeVLongArray(responseTimeHistogram);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        HttpRouteStats that = (HttpRouteStats) o;
+        return requestCount == that.requestCount
+            && totalRequestSize == that.totalRequestSize
+            && responseCount == that.responseCount
+            && totalResponseSize == that.totalResponseSize
+            && Arrays.equals(requestSizeHistogram, that.requestSizeHistogram)
+            && Arrays.equals(responseSizeHistogram, that.responseSizeHistogram)
+            && Arrays.equals(responseTimeHistogram, that.responseTimeHistogram);
+    }
+
+    @Override
+    public int hashCode() {
+        int result = Objects.hash(requestCount, totalRequestSize, responseCount, totalResponseSize);
+        result = 31 * result + Arrays.hashCode(requestSizeHistogram);
+        result = 31 * result + Arrays.hashCode(responseSizeHistogram);
+        result = 31 * result + Arrays.hashCode(responseTimeHistogram);
+        return result;
+    }
```

Comment:
This is a record, do we need override equals and hashCode? Or they do not handle arrays?
In such case would it be beneficial to wrap array into Histogram object?
This could also simplify merging them and serializing toXContent